### DISCUSSION TITLE
Navigation / Sitelinks enhancement

### DISCUSSION THREAD

**Discussion by Jimmi08** [e107inc/e107#4690] [Ideas]
Add custom fields to sitelinks.  
Add sitelinks categories to sitelinks (alternative to hardcoded types - main, alt...)

**Categories**
They don't need to be editable, just custom fields for them. But adding template there, you would avoid hardcoded template in NAVIGATION shortcode.  But this is just side effect, main goal are custom fields for links (and for this you need categories - at least for now).  

**Custom Fields**
With advanced navigation just templating is not enough. 

Example:
this is my needed navigation:
![image](https://user-images.githubusercontent.com/5429548/153378215-542c640b-3120-4e50-86e7-ef3f09cf23d1.png)

after click on page
 
![image](https://user-images.githubusercontent.com/5429548/153379038-52bb962f-762b-49f0-9381-42626c34d3f8.png)

to get this, those custom fields are needed for each link

```
			<field name="page_fields">{
    "accent": "cyan-dark",
    "navbgimage": "{e_THEME}boots5\/images\/navigation\/contact-6-3.png",
    "navsmimage": "{e_THEME}boots5\/images\/lineicons\/map-6-3.png",    
    "smallimage": "{e_THEME}boots5\/images\/lineicons\/map.png",
    "sideimage": "{e_THEME}boots5\/images\/sidebars\/contact.png",
    "pageimage": "{e_THEME}boots5\/images\/contact_bg.png",
    "bggradient":   "soft-orange-1",
    "color": "primary"
}</field>
```

It would be really nice if sitelinks have the same functionality as chapter/pages

Everything is already there. Just use the same functionality for sitelinks like for pages. 

Thanks

**Comment by tgtje** [comment]
Hmm i can see where you are going to, seems to be a good enhancement, just wondering if it not to theme builder 'property'.
Would like to see a working site, since this (imo) is the 'next' step in 'difference' from those website 'look all the same'. 

:dart:

**Reply by Jimmi08** [reply]
you can see the bootstrap4 version here:  https://caymandemo.e107sk.com/   - but it will be replaced soon.

Pages are used for navigation. But how do you add contact.php or news pages etc?  Standalone e107 pages?

custom fields handler is already built-in e107, you just need to have some parent for setting it.

I asked this for featurebox - denied because ... no comment.

I tried to build a plugin for this... e_admin is not working (or I didn't know how to fix it) - to extend existing admin_ui .  I asked for a basic example - some simple plugin version...  

I asked to add some keys to sitelinks, not just id, just key - so you are able to set this in theme prefs. Because Id is changing for installation, but the key (name) is stable. 

PS. Bootstrap 5 version changes footer color (classes) related to selected page/link...  but you can't do it without link identification, and ID is just not enough...

**Comment by Jimmi08** [comment]
@tgtje on second thought ...  if the theme was able to define custom fields for sitelinks (like chapters) and they will be displayed similar way as e_admin work, it could solve this problem. I like that idea. Then link categories are not needed then.  
Thanks
